Add a noop chaincode using TxID instead of UUID

gabre opened the pull request:
This PR introduces a trivial chain code, that serves a subset of the fabric_api, namely sendTransaction, getTransaction and listen for transactions.
## Description

A wider reaching change to devops.go is that "invoke" type transactions will be assigned an id derived from their user provided content instead of a random UUID. This is needed, so a client submitting a transaction is able to check if it ends up being in the block. The change makes however following assumptions, that potentially break current use:

1) User provided transaction content of "invoke" type transactions is unique
2) User provided transaction content has a known encoding (currently base64)
## Motivation and Context

Begin to build a chaincode serving fabric_api
## How Has This Been Tested?

Unit tests were run. Behavior tests were run.
## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Gabor Hosszu (gabor@digitalasset.com)

gaborh-da committed:
Add noop system chaincode

gaborh-da committed:
Use payload hash instead of UUID

gaborh-da committed:
Use shim logging

gaborh-da committed:
Remove Base64 decoding of input in noop syscc

gaborh-da committed:
Implement getTran functionality for noop syscc

gaborh-da committed:
Use only user given data for txID (hashing)

gaborh-da committed:
Standardize data types in noop syscc

Input of Noop Exec is Base64 Str. This input is decoded and passed
to Noop internal logic functions as a byte array.
Fabric internally stores this input as Base64 Str. In Noop Query,
we decode this and the byte array form is returned.

gaborh-da committed:
Handle query and invoke TxIDs differently

gaborh-da committed:
Remove old utxo tests from noop directory

Old tests copied from another chaincode were removed.

gaborh-da committed:
Remove unnecessary files from directory of noop syscc

tamasblummer commented on core/devops.go:
```diff
@@ -208,37 +209,55 @@ func (d *Devops) invokeOrQuery(ctx context.Context, chaincodeInvocationSpec *pb.
+func (d *Devops) createExecTx(spec *pb.ChaincodeInvocationSpec, invokeTx bool, sec crypto.Client) (*pb.Transaction, error) {
 	var tx *pb.Transaction
 	var err error
+        var temp_uuid = ""
```
why using temp_uuid instead of calculated?

gabre replied:
You are right, there is no reason to use that. We can use the calculated as it was before the changes. I will upload a fix.

gaborh-da committed:
Move TxID generation to its old place (UUID generation)

TxID double hash generation was moved to the same place where
UUID was generated before the ID got the traditional BTC TxID
instead of UUID. There is no need for temporary IDs so they
were removed.

gaborh-da committed:
Use UUID instead of the same TxID for queries

For queries, we used TxID generated for a byte array full of zero
bytes. Now we use a really unique UUID.

gaborh-da committed:
Remove old comment leftover

gaborh-da committed:
Fix incorrect ID generation for queries

tamasblummer commented:
Change to TXID will rather be implemented with an optional parameter to "invoke" type transactions
